Using the move constructor of `small::vector` fails to build on FreeBSD

## The problem

If the move constructor of `small::vector` is used, the program will fail to build on FreeBSD.

It could be a problem with the llvm headers, I'm not sure yet.

### Steps to Reproduce

<details>
<summary>CMakeLists.txt</summary>

```cmake
cmake_minimum_required(VERSION 3.22)
project(small_ftbfs)

set(CMAKE_CXX_STANDARD 17)

include(FetchContent)

FetchContent_Declare(small
        GIT_REPOSITORY https://github.com/alandefreitas/small
        GIT_TAG origin/master
)

FetchContent_GetProperties(small)
if(NOT small_POPULATED)
    FetchContent_Populate(small)
    add_subdirectory(${small_SOURCE_DIR} ${small_BINARY_DIR} EXCLUDE_FROM_ALL)
endif()

add_executable(small_ftbfs main.cpp)
target_link_libraries(small_ftbfs PUBLIC small::small)
```

</details>

```c++
#include <iostream>
#include <utility>

#include <small/vector.h>

using Vec = small::max_size_vector<std::pair<int, int>>;

Vec create()
{
    auto ret = Vec{};
    ret.emplace_back(1, 2);
    ret.emplace_back(3, 4);
    return ret;
}

int main()
{
    auto vec = Vec{};
    vec = create();

    for (auto [lhs, rhs] : vec)
    {
        std::cout << lhs << ' ' << rhs << std::endl;
    }

    return 0;
}

```

Note: You need to patch `vector.hpp` with #30 first, or else this problem will get masked by it.

### Output

<details>
 
```console
In file included from /root/small-ftbfs/main.cpp:1:
In file included from /usr/include/c++/v1/iostream:37:
In file included from /usr/include/c++/v1/ios:214:
In file included from /usr/include/c++/v1/__locale:18:
In file included from /usr/include/c++/v1/memory:819:
In file included from /usr/include/c++/v1/__memory/ranges_uninitialized_algorithms.h:22:
/usr/include/c++/v1/__memory/uninitialized_algorithms.h:45:41: error: use of overloaded operator '!=' is ambiguous (with operand types 'small::pointer_wrapper<std::pair<int, int> *>' and 'std::__unreachable_sentinel')
    for (; __ifirst != __ilast && __idx != __olast; ++__ifirst, (void)++__idx)
                                  ~~~~~ ^  ~~~~~~~
/usr/include/c++/v1/__memory/uninitialized_algorithms.h:61:26: note: in instantiation of function template specialization 'std::__uninitialized_copy<std::pair<int, int>, std::move_iterator<small::pointer_wrapper<std::pair<int, int> *>>, std::move_iterator<small::pointer_wrapper<std::pair<int, int> *>>, small::pointer_wrapper<std::pair<int, int> *>, std::__unreachable_sentinel>' requested here
  auto __result = _VSTD::__uninitialized_copy<_ValueType>(_VSTD::move(__ifirst), _VSTD::move(__ilast),
                         ^
/root/small-ftbfs/build_deb/_deps/small-src/source/small/vector.h:241:30: note: in instantiation of function template specialization 'std::uninitialized_copy<std::move_iterator<small::pointer_wrapper<std::pair<int, int> *>>, small::pointer_wrapper<std::pair<int, int> *>>' requested here
                        std::uninitialized_copy(std::make_move_iterator(rhs.begin()),
                             ^
/root/small-ftbfs/main.cpp:13:12: note: in instantiation of member function 'small::vector<std::pair<int, int>, 6, std::allocator<std::pair<int, int>>, std::integral_constant<bool, false>>::vector' requested here
    return ret;
           ^
/usr/include/c++/v1/__memory/uninitialized_algorithms.h:30:55: note: candidate function [with _Iter = small::pointer_wrapper<std::pair<int, int> *>]
  _LIBCPP_HIDE_FROM_ABI friend _LIBCPP_CONSTEXPR bool operator!=(const _Iter&, __unreachable_sentinel) _NOEXCEPT {
                                                      ^
/root/small-ftbfs/build_deb/_deps/small-src/source/small/detail/iterator/pointer_wrapper.h:182:27: note: candidate function [with Iter = std::pair<int, int> *, BaseIter = std::__unreachable_sentinel]
    inline constexpr bool operator!=(const pointer_wrapper<Iter> &x, const BaseIter &y) noexcept {
```
</details>

## Platform
<!-- Mark the platforms where the bug happens -->

- [ ] *cross-platform issue - linux*
- [ ] *cross-platform issue - windows*
- [ ] *cross-platform issue - macos*
- [x] *other:* FreeBSD 13.2

### Environment Details
 - OS: FreeBSD
 - OS Version: 13.2-RELEASE
 - Compiler: clang included with FreeBSD
 - Compiler version: 14.0.5

## Proposed solution

## Alternatives I've considered

### Additional context
<!--optional-->

Hmm, I wonder why I got `vector.h` instead of `vector.hpp`... I just followed the "Fetch content" `CMakeLists.txt` example on the README...

Anyway, here's the output using `vector.hpp` instead.

```console
In file included from /root/small-ftbfs/main.cpp:1:
In file included from /usr/include/c++/v1/iostream:37:
In file included from /usr/include/c++/v1/ios:214:
In file included from /usr/include/c++/v1/__locale:18:
In file included from /usr/include/c++/v1/memory:819:
In file included from /usr/include/c++/v1/__memory/ranges_uninitialized_algorithms.h:22:
/usr/include/c++/v1/__memory/uninitialized_algorithms.h:45:41: error: use of overloaded operator '!=' is ambiguous (with operand types 'small::detail::pointer_wrapper<std::pair<int, int> *>' and 'std::__unreachable_sentinel')
    for (; __ifirst != __ilast && __idx != __olast; ++__ifirst, (void)++__idx)
                                  ~~~~~ ^  ~~~~~~~
/usr/include/c++/v1/__memory/uninitialized_algorithms.h:61:26: note: in instantiation of function template specialization 'std::__uninitialized_copy<std::pair<int, int>, std::move_iterator<small::detail::pointer_wrapper<std::pair<int, int> *>>, std::move_iterator<small::detail::pointer_wrapper<std::pair<int, int> *>>, small::detail::pointer_wrapper<std::pair<int, int> *>, std::__unreachable_sentinel>' requested here
  auto __result = _VSTD::__uninitialized_copy<_ValueType>(_VSTD::move(__ifirst), _VSTD::move(__ilast),
                         ^
/root/small-ftbfs/small/include/small/vector.hpp:292:30: note: in instantiation of function template specialization 'std::uninitialized_copy<std::move_iterator<small::detail::pointer_wrapper<std::pair<int, int> *>>, small::detail::pointer_wrapper<std::pair<int, int> *>>' requested here
                        std::uninitialized_copy(
                             ^
/root/small-ftbfs/main.cpp:13:12: note: in instantiation of member function 'small::vector<std::pair<int, int>, 6, std::allocator<std::pair<int, int>>, std::integral_constant<bool, false>>::vector' requested here
    return ret;
           ^
/usr/include/c++/v1/__memory/uninitialized_algorithms.h:30:55: note: candidate function [with _Iter = small::detail::pointer_wrapper<std::pair<int, int> *>]
  _LIBCPP_HIDE_FROM_ABI friend _LIBCPP_CONSTEXPR bool operator!=(const _Iter&, __unreachable_sentinel) _NOEXCEPT {
                                                      ^
/root/small-ftbfs/small/include/small/detail/iterator/pointer_wrapper.hpp:263:9: note: candidate function [with Iter = std::pair<int, int> *, BaseIter = std::__unreachable_sentinel]
        operator!=(const pointer_wrapper<Iter> &x, const BaseIter &y) noexcept {
        ^
1 error generated.
```

--- a/include/small/detail/iterator/pointer_wrapper.hpp
+++ b/include/small/detail/iterator/pointer_wrapper.hpp
@@ -252,63 +252,63 @@ namespace small {
             return !(y < x);
         }
 
-        template <class Iter, class BaseIter>
+        template <class Iter>
         inline constexpr bool
-        operator==(const pointer_wrapper<Iter> &x, const BaseIter &y) noexcept {
+        operator==(const pointer_wrapper<Iter> &x, Iter y) noexcept {
             return x.base() == y;
         }
 
-        template <class Iter, class BaseIter>
+        template <class Iter>
         inline constexpr bool
-        operator!=(const pointer_wrapper<Iter> &x, const BaseIter &y) noexcept {
+        operator!=(const pointer_wrapper<Iter> &x, Iter y) noexcept {
             return !(x == y);
         }
 
-        template <class Iter, class BaseIter>
+        template <class Iter>
         inline constexpr bool
-        operator>(const pointer_wrapper<Iter> &x, const BaseIter &y) noexcept {
+        operator>(const pointer_wrapper<Iter> &x, Iter y) noexcept {
             return y < x;
         }
 
-        template <class Iter, class BaseIter>
+        template <class Iter>
         inline constexpr bool
-        operator>=(const pointer_wrapper<Iter> &x, const BaseIter &y) noexcept {
+        operator>=(const pointer_wrapper<Iter> &x, Iter y) noexcept {
             return !(x < y);
         }
 
-        template <class Iter, class BaseIter>
+        template <class Iter>
         inline constexpr bool
-        operator<=(const pointer_wrapper<Iter> &x, const BaseIter &y) noexcept {
+        operator<=(const pointer_wrapper<Iter> &x, Iter y) noexcept {
             return !(y < x);
         }
 
-        template <class Iter, class BaseIter>
+        template <class Iter>
         inline constexpr bool
-        operator==(const BaseIter &x, const pointer_wrapper<Iter> &y) noexcept {
+        operator==(Iter x, const pointer_wrapper<Iter> &y) noexcept {
             return x.base() == y.base();
         }
 
-        template <class Iter, class BaseIter>
+        template <class Iter>
         inline constexpr bool
-        operator!=(const BaseIter &x, const pointer_wrapper<Iter> &y) noexcept {
+        operator!=(Iter x, const pointer_wrapper<Iter> &y) noexcept {
             return !(x == y);
         }
 
-        template <class Iter, class BaseIter>
+        template <class Iter>
         inline constexpr bool
-        operator>(const BaseIter &x, const pointer_wrapper<Iter> &y) noexcept {
+        operator>(Iter x, const pointer_wrapper<Iter> &y) noexcept {
             return y < x;
         }
 
-        template <class Iter, class BaseIter>
+        template <class Iter>
         inline constexpr bool
-        operator>=(const BaseIter &x, const pointer_wrapper<Iter> &y) noexcept {
+        operator>=(Iter x, const pointer_wrapper<Iter> &y) noexcept {
             return !(x < y);
         }
 
-        template <class Iter, class BaseIter>
+        template <class Iter>
         inline constexpr bool
-        operator<=(const BaseIter &x, const pointer_wrapper<Iter> &y) noexcept {
+        operator<=(Iter x, const pointer_wrapper<Iter> &y) noexcept {
             return !(y < x);
         }
 

--- a/include/small/detail/iterator/pointer_wrapper.hpp
+++ b/include/small/detail/iterator/pointer_wrapper.hpp
@@ -146,186 +146,186 @@ namespace small {
 
             public /* friends */:
             /// \brief Get distance between iterators
-            template <class Iter1, class Iter2>
+            template <class Pointer1, class Pointer2>
             constexpr friend auto
             operator-(
-                const pointer_wrapper<Iter1> &x,
-                const pointer_wrapper<Iter2> &y) noexcept
+                const pointer_wrapper<Pointer1> &x,
+                const pointer_wrapper<Pointer2> &y) noexcept
                 -> decltype(x.base() - y.base());
 
             /// \brief Sum iterators
-            template <class Iter1>
-            constexpr friend pointer_wrapper<Iter1> operator+(
-                typename pointer_wrapper<Iter1>::difference_type,
-                pointer_wrapper<Iter1>) noexcept;
+            template <class Pointer1>
+            constexpr friend pointer_wrapper<Pointer1> operator+(
+                typename pointer_wrapper<Pointer1>::difference_type,
+                pointer_wrapper<Pointer1>) noexcept;
 
         private:
             /// Base pointer
             iterator_type base_;
         };
 
-        template <class Iter1, class Iter2>
+        template <class Pointer1, class Pointer2>
         inline constexpr bool
         operator==(
-            const pointer_wrapper<Iter1> &x,
-            const pointer_wrapper<Iter2> &y) noexcept {
+            const pointer_wrapper<Pointer1> &x,
+            const pointer_wrapper<Pointer2> &y) noexcept {
             return x.base() == y.base();
         }
 
-        template <class Iter1, class Iter2>
+        template <class Pointer1, class Pointer2>
         inline constexpr bool
         operator!=(
-            const pointer_wrapper<Iter1> &x,
-            const pointer_wrapper<Iter2> &y) noexcept {
+            const pointer_wrapper<Pointer1> &x,
+            const pointer_wrapper<Pointer2> &y) noexcept {
             return !(x == y);
         }
 
-        template <class Iter1, class Iter2>
+        template <class Pointer1, class Pointer2>
         inline constexpr bool
         operator<(
-            const pointer_wrapper<Iter1> &x,
-            const pointer_wrapper<Iter2> &y) noexcept {
+            const pointer_wrapper<Pointer1> &x,
+            const pointer_wrapper<Pointer2> &y) noexcept {
             return x.base() < y.base();
         }
 
-        template <class Iter1, class Iter2>
+        template <class Pointer1, class Pointer2>
         inline constexpr bool
         operator>(
-            const pointer_wrapper<Iter1> &x,
-            const pointer_wrapper<Iter2> &y) noexcept {
+            const pointer_wrapper<Pointer1> &x,
+            const pointer_wrapper<Pointer2> &y) noexcept {
             return y < x;
         }
 
-        template <class Iter1, class Iter2>
+        template <class Pointer1, class Pointer2>
         inline constexpr bool
         operator>=(
-            const pointer_wrapper<Iter1> &x,
-            const pointer_wrapper<Iter2> &y) noexcept {
+            const pointer_wrapper<Pointer1> &x,
+            const pointer_wrapper<Pointer2> &y) noexcept {
             return !(x < y);
         }
 
-        template <class Iter1, class Iter2>
+        template <class Pointer1, class Pointer2>
         inline constexpr bool
         operator<=(
-            const pointer_wrapper<Iter1> &x,
-            const pointer_wrapper<Iter2> &y) noexcept {
+            const pointer_wrapper<Pointer1> &x,
+            const pointer_wrapper<Pointer2> &y) noexcept {
             return !(y < x);
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
         operator==(
-            const pointer_wrapper<Iter> &x,
-            const pointer_wrapper<Iter> &y) noexcept {
+            const pointer_wrapper<Pointer> &x,
+            const pointer_wrapper<Pointer> &y) noexcept {
             return x.base() == y.base();
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
         operator!=(
-            const pointer_wrapper<Iter> &x,
-            const pointer_wrapper<Iter> &y) noexcept {
+            const pointer_wrapper<Pointer> &x,
+            const pointer_wrapper<Pointer> &y) noexcept {
             return !(x == y);
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
         operator>(
-            const pointer_wrapper<Iter> &x,
-            const pointer_wrapper<Iter> &y) noexcept {
+            const pointer_wrapper<Pointer> &x,
+            const pointer_wrapper<Pointer> &y) noexcept {
             return y < x;
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
         operator>=(
-            const pointer_wrapper<Iter> &x,
-            const pointer_wrapper<Iter> &y) noexcept {
+            const pointer_wrapper<Pointer> &x,
+            const pointer_wrapper<Pointer> &y) noexcept {
             return !(x < y);
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
         operator<=(
-            const pointer_wrapper<Iter> &x,
-            const pointer_wrapper<Iter> &y) noexcept {
+            const pointer_wrapper<Pointer> &x,
+            const pointer_wrapper<Pointer> &y) noexcept {
             return !(y < x);
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
-        operator==(const pointer_wrapper<Iter> &x, Iter y) noexcept {
+        operator==(const pointer_wrapper<Pointer> &x, Pointer y) noexcept {
             return x.base() == y;
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
-        operator!=(const pointer_wrapper<Iter> &x, Iter y) noexcept {
+        operator!=(const pointer_wrapper<Pointer> &x, Pointer y) noexcept {
             return !(x == y);
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
-        operator>(const pointer_wrapper<Iter> &x, Iter y) noexcept {
+        operator>(const pointer_wrapper<Pointer> &x, Pointer y) noexcept {
             return y < x;
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
-        operator>=(const pointer_wrapper<Iter> &x, Iter y) noexcept {
+        operator>=(const pointer_wrapper<Pointer> &x, Pointer y) noexcept {
             return !(x < y);
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
-        operator<=(const pointer_wrapper<Iter> &x, Iter y) noexcept {
+        operator<=(const pointer_wrapper<Pointer> &x, Pointer y) noexcept {
             return !(y < x);
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
-        operator==(Iter x, const pointer_wrapper<Iter> &y) noexcept {
+        operator==(Pointer x, const pointer_wrapper<Pointer> &y) noexcept {
             return x.base() == y.base();
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
-        operator!=(Iter x, const pointer_wrapper<Iter> &y) noexcept {
+        operator!=(Pointer x, const pointer_wrapper<Pointer> &y) noexcept {
             return !(x == y);
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
-        operator>(Iter x, const pointer_wrapper<Iter> &y) noexcept {
+        operator>(Pointer x, const pointer_wrapper<Pointer> &y) noexcept {
             return y < x;
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
-        operator>=(Iter x, const pointer_wrapper<Iter> &y) noexcept {
+        operator>=(Pointer x, const pointer_wrapper<Pointer> &y) noexcept {
             return !(x < y);
         }
 
-        template <class Iter>
+        template <class Pointer>
         inline constexpr bool
-        operator<=(Iter x, const pointer_wrapper<Iter> &y) noexcept {
+        operator<=(Pointer x, const pointer_wrapper<Pointer> &y) noexcept {
             return !(y < x);
         }
 
-        template <class Iter1, class Iter2>
+        template <class Pointer1, class Pointer2>
         inline constexpr auto
         operator-(
-            const pointer_wrapper<Iter1> &x,
-            const pointer_wrapper<Iter2> &y) noexcept
+            const pointer_wrapper<Pointer1> &x,
+            const pointer_wrapper<Pointer2> &y) noexcept
             -> decltype(x.base() - y.base()) {
             return x.base() - y.base();
         }
 
-        template <class Iter>
-        inline constexpr pointer_wrapper<Iter>
+        template <class Pointer>
+        inline constexpr pointer_wrapper<Pointer>
         operator+(
-            typename pointer_wrapper<Iter>::difference_type n,
-            pointer_wrapper<Iter> x) noexcept {
+            typename pointer_wrapper<Pointer>::difference_type n,
+            pointer_wrapper<Pointer> x) noexcept {
             x += n;
             return x;
         }

--- a/include/small/detail/iterator/pointer_wrapper.hpp
+++ b/include/small/detail/iterator/pointer_wrapper.hpp
@@ -165,7 +165,7 @@ namespace small {
             iterator_type base_;
         };
 
-        template <class Pointer1, class Pointer2>
+        template <class Pointer1, class Pointer2 = Pointer1>
         inline constexpr bool
         operator==(
             const pointer_wrapper<Pointer1> &x,
@@ -173,7 +173,7 @@ namespace small {
             return x.base() == y.base();
         }
 
-        template <class Pointer1, class Pointer2>
+        template <class Pointer1, class Pointer2 = Pointer1>
         inline constexpr bool
         operator!=(
             const pointer_wrapper<Pointer1> &x,
@@ -181,7 +181,7 @@ namespace small {
             return !(x == y);
         }
 
-        template <class Pointer1, class Pointer2>
+        template <class Pointer1, class Pointer2 = Pointer1>
         inline constexpr bool
         operator<(
             const pointer_wrapper<Pointer1> &x,
@@ -189,7 +189,7 @@ namespace small {
             return x.base() < y.base();
         }
 
-        template <class Pointer1, class Pointer2>
+        template <class Pointer1, class Pointer2 = Pointer1>
         inline constexpr bool
         operator>(
             const pointer_wrapper<Pointer1> &x,
@@ -197,15 +197,7 @@ namespace small {
             return y < x;
         }
 
-        template <class Pointer1, class Pointer2>
-        inline constexpr bool
-        operator>=(
-            const pointer_wrapper<Pointer1> &x,
-            const pointer_wrapper<Pointer2> &y) noexcept {
-            return !(x < y);
-        }
-
-        template <class Pointer1, class Pointer2>
+        template <class Pointer1, class Pointer2 = Pointer1>
         inline constexpr bool
         operator<=(
             const pointer_wrapper<Pointer1> &x,
@@ -213,44 +205,12 @@ namespace small {
             return !(y < x);
         }
 
-        template <class Pointer>
-        inline constexpr bool
-        operator==(
-            const pointer_wrapper<Pointer> &x,
-            const pointer_wrapper<Pointer> &y) noexcept {
-            return x.base() == y.base();
-        }
-
-        template <class Pointer>
-        inline constexpr bool
-        operator!=(
-            const pointer_wrapper<Pointer> &x,
-            const pointer_wrapper<Pointer> &y) noexcept {
-            return !(x == y);
-        }
-
-        template <class Pointer>
-        inline constexpr bool
-        operator>(
-            const pointer_wrapper<Pointer> &x,
-            const pointer_wrapper<Pointer> &y) noexcept {
-            return y < x;
-        }
-
-        template <class Pointer>
+        template <class Pointer1, class Pointer2 = Pointer1>
         inline constexpr bool
         operator>=(
-            const pointer_wrapper<Pointer> &x,
-            const pointer_wrapper<Pointer> &y) noexcept {
+            const pointer_wrapper<Pointer1> &x,
+            const pointer_wrapper<Pointer2> &y) noexcept {
             return !(x < y);
-        }
-
-        template <class Pointer>
-        inline constexpr bool
-        operator<=(
-            const pointer_wrapper<Pointer> &x,
-            const pointer_wrapper<Pointer> &y) noexcept {
-            return !(y < x);
         }
 
         template <class Pointer>
@@ -271,18 +231,18 @@ namespace small {
 
         template <class Pointer>
         inline constexpr bool
-        operator>(
+        operator<(
             const pointer_wrapper<Pointer> &x,
             ptr_to_const_t<Pointer> y) noexcept {
-            return y < x;
+            return x.base() < y;
         }
 
         template <class Pointer>
         inline constexpr bool
-        operator>=(
+        operator>(
             const pointer_wrapper<Pointer> &x,
             ptr_to_const_t<Pointer> y) noexcept {
-            return !(x < y);
+            return y < x.base();
         }
 
         template <class Pointer>
@@ -295,10 +255,18 @@ namespace small {
 
         template <class Pointer>
         inline constexpr bool
+        operator>=(
+            const pointer_wrapper<Pointer> &x,
+            ptr_to_const_t<Pointer> y) noexcept {
+            return !(x < y);
+        }
+
+        template <class Pointer>
+        inline constexpr bool
         operator==(
             ptr_to_const_t<Pointer> x,
             const pointer_wrapper<Pointer> &y) noexcept {
-            return x.base() == y.base();
+            return x == y.base();
         }
 
         template <class Pointer>
@@ -319,10 +287,10 @@ namespace small {
 
         template <class Pointer>
         inline constexpr bool
-        operator>=(
+        operator<(
             ptr_to_const_t<Pointer> x,
             const pointer_wrapper<Pointer> &y) noexcept {
-            return !(x < y);
+            return y > x;
         }
 
         template <class Pointer>
@@ -333,7 +301,15 @@ namespace small {
             return !(y < x);
         }
 
-        template <class Pointer1, class Pointer2>
+        template <class Pointer>
+        inline constexpr bool
+        operator>=(
+            ptr_to_const_t<Pointer> x,
+            const pointer_wrapper<Pointer> &y) noexcept {
+            return !(x < y);
+        }
+
+        template <class Pointer1, class Pointer2 = Pointer1>
         inline constexpr auto
         operator-(
             const pointer_wrapper<Pointer1> &x,

--- a/include/small/detail/iterator/pointer_wrapper.hpp
+++ b/include/small/detail/iterator/pointer_wrapper.hpp
@@ -8,6 +8,7 @@
 #ifndef SMALL_DETAIL_ITERATOR_POINTER_WRAPPER_HPP
 #define SMALL_DETAIL_ITERATOR_POINTER_WRAPPER_HPP
 
+#include <small/detail/traits/ptr_to_const.hpp>
 #include <algorithm>
 #include <iterator>
 #include <type_traits>
@@ -254,61 +255,81 @@ namespace small {
 
         template <class Pointer>
         inline constexpr bool
-        operator==(const pointer_wrapper<Pointer> &x, Pointer y) noexcept {
+        operator==(
+            const pointer_wrapper<Pointer> &x,
+            ptr_to_const_t<Pointer> y) noexcept {
             return x.base() == y;
         }
 
         template <class Pointer>
         inline constexpr bool
-        operator!=(const pointer_wrapper<Pointer> &x, Pointer y) noexcept {
+        operator!=(
+            const pointer_wrapper<Pointer> &x,
+            ptr_to_const_t<Pointer> y) noexcept {
             return !(x == y);
         }
 
         template <class Pointer>
         inline constexpr bool
-        operator>(const pointer_wrapper<Pointer> &x, Pointer y) noexcept {
+        operator>(
+            const pointer_wrapper<Pointer> &x,
+            ptr_to_const_t<Pointer> y) noexcept {
             return y < x;
         }
 
         template <class Pointer>
         inline constexpr bool
-        operator>=(const pointer_wrapper<Pointer> &x, Pointer y) noexcept {
+        operator>=(
+            const pointer_wrapper<Pointer> &x,
+            ptr_to_const_t<Pointer> y) noexcept {
             return !(x < y);
         }
 
         template <class Pointer>
         inline constexpr bool
-        operator<=(const pointer_wrapper<Pointer> &x, Pointer y) noexcept {
+        operator<=(
+            const pointer_wrapper<Pointer> &x,
+            ptr_to_const_t<Pointer> y) noexcept {
             return !(y < x);
         }
 
         template <class Pointer>
         inline constexpr bool
-        operator==(Pointer x, const pointer_wrapper<Pointer> &y) noexcept {
+        operator==(
+            ptr_to_const_t<Pointer> x,
+            const pointer_wrapper<Pointer> &y) noexcept {
             return x.base() == y.base();
         }
 
         template <class Pointer>
         inline constexpr bool
-        operator!=(Pointer x, const pointer_wrapper<Pointer> &y) noexcept {
+        operator!=(
+            ptr_to_const_t<Pointer> x,
+            const pointer_wrapper<Pointer> &y) noexcept {
             return !(x == y);
         }
 
         template <class Pointer>
         inline constexpr bool
-        operator>(Pointer x, const pointer_wrapper<Pointer> &y) noexcept {
+        operator>(
+            ptr_to_const_t<Pointer> x,
+            const pointer_wrapper<Pointer> &y) noexcept {
             return y < x;
         }
 
         template <class Pointer>
         inline constexpr bool
-        operator>=(Pointer x, const pointer_wrapper<Pointer> &y) noexcept {
+        operator>=(
+            ptr_to_const_t<Pointer> x,
+            const pointer_wrapper<Pointer> &y) noexcept {
             return !(x < y);
         }
 
         template <class Pointer>
         inline constexpr bool
-        operator<=(Pointer x, const pointer_wrapper<Pointer> &y) noexcept {
+        operator<=(
+            ptr_to_const_t<Pointer> x,
+            const pointer_wrapper<Pointer> &y) noexcept {
             return !(y < x);
         }
 

--- a/include/small/detail/traits/ptr_to_const.hpp
+++ b/include/small/detail/traits/ptr_to_const.hpp
@@ -1,0 +1,25 @@
+//
+// Copyright (c) 2024 Yat Ho (lagoho7@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
+//
+
+#ifndef SMALL_DETAIL_TRAITS_PTR_TO_CONST_HPP
+#define SMALL_DETAIL_TRAITS_PTR_TO_CONST_HPP
+
+#include <type_traits>
+
+namespace small {
+    namespace detail {
+        /// Convert a pointer to pointer-to-const
+        template <typename T, typename = std::enable_if_t<std::is_pointer_v<T>>>
+        using ptr_to_const = std::add_pointer<
+            std::add_const_t<std::remove_pointer_t<T>>>;
+
+        template <typename T>
+        using ptr_to_const_t = typename ptr_to_const<T>::type;
+    } // namespace detail
+} // namespace small
+
+#endif // SMALL_DETAIL_TRAITS_PTR_TO_CONST_HPP

--- a/test/unit/string_small_vector.cpp
+++ b/test/unit/string_small_vector.cpp
@@ -206,6 +206,36 @@ TEST_CASE("String Vector") {
 
     SECTION("Iterators") {
         vector<std::string, 5> a = { "one", "two", "three" };
+        auto iter = a.begin();
+
+        // LegacyContiguousIterator
+        REQUIRE(*(a.begin() + 2) == *(std::addressof(*a.begin()) + 2));
+
+        // LegacyRandomAccessIterator
+        REQUIRE((iter += 2) == a.begin() + 2);
+        REQUIRE(a.begin() + 1 == 1 + a.begin());
+        REQUIRE((iter -= 2) == a.begin());
+        REQUIRE(a.end() - 2 == a.begin() + 1);
+        REQUIRE(a.end() - a.begin() == 3);
+        REQUIRE(a.begin()[1] == "two");
+        REQUIRE(a.end()[-1] == "three");
+
+        REQUIRE(a.begin() == a.begin());
+        REQUIRE(a.begin() != a.begin() + 2);
+        REQUIRE(a.begin() < a.begin() + 2);
+        REQUIRE(a.begin() + 2 > a.begin());
+
+        // LegacyBidirectionalIterator + LegacyForwardIterator
+        iter = a.begin() + 1;
+        REQUIRE(--iter == a.begin());
+        REQUIRE(iter++ == a.begin());
+        REQUIRE(iter-- == a.begin() + 1);
+        iter = a.begin() + 1;
+        REQUIRE(*iter-- == "two");
+
+        // LegacyInputIterator/LegacyIterator
+        iter = a.begin();
+        REQUIRE(++iter == a.begin() + 1);
 
         REQUIRE(a.begin() == a.data());
         REQUIRE(a.end() == a.data() + a.size());
@@ -224,6 +254,17 @@ TEST_CASE("String Vector") {
 
         REQUIRE(*a.crbegin() == "three");
         REQUIRE(*std::prev(a.crend()) == "one");
+
+        // Custom comparison operators
+        REQUIRE(a.begin() == a.data());
+        REQUIRE(a.begin() != a.data() + 2);
+        REQUIRE(a.begin() < a.data() + 2);
+        REQUIRE(a.begin() + 2 > a.data());
+
+        REQUIRE(a.data() == a.begin());
+        REQUIRE(a.data() != a.begin() + 2);
+        REQUIRE(a.data() < a.begin() + 2);
+        REQUIRE(a.data() + 2 > a.begin());
     }
 
     SECTION("Capacity") {

